listenbrainz/webserver/views/playlist_api.py

```diff
@@ -444,10 +444,12 @@ def get_playlist(playlist_mbid):
     if playlist is None:
         raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
 
-    if not playlist.public:
-        user = validate_auth_header()
-        if playlist.creator_id != user["id"]:
-            raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
+    user = validate_auth_header(True)
+    user_id = None
+    if user:
+        user_id = user["id"]
```

Comment:
The user is optional here but required in the other endpoints, and I wanted to ensure we call `is_visible_by` with an existing user id, but I suppose it can be optional (as you suggested in another review comment).
Will change it accordingly.